File: source/TaskBoard.Web/src/app/models/card.ts
export class Card {
    public id: number;
    public name: string;
    public priorityId: number;
    public dueDate: Date;
    public description: string;
    public statusId: number;
    public priorityName: string;


    constructor(card: Omit<Card, 'id' | 'priorityName'>);
    
    constructor(card: Omit<Card, 'priorityName'>)

    constructor(card: Omit<Card, 'id' | 'priorityName'> | Omit<Card, 'priorityName'>) {
        this.name = card.name;
        this.priorityId = card.priorityId;
        this.dueDate = card.dueDate;
        this.description = card.description;
        this.statusId = card.statusId;

        if('id' in card){
            this.id = card.id;
        }
    }
}

File: source/TaskBoard.DAL/src/Entities/Card.cs
namespace TaskBoard.DAL.Entities;

public class Card : BaseEntity
{
	public string Name { get; set; }
	public int StatusId { get; set; }
	public DateTime DueDate { get; set; }
	public int PriorityId { get; set; }
	public string Description { get; set; }
	
	public Status Status { get; set; } 
	public Priority Priority { get; set; }
}